### PR TITLE
Fix the Android build and improve the Android building `README.md`

### LINKED ISSUE #6321: Android build failed
[error.txt](https://github.com/ddnet/ddnet/files/10553516/error.txt)

### PR DESCRIPTION
**Pull request by Robyt3**
This fixes the Android build, which involves

- adding Rust support to Android compilation,
- upgrading Gradle and fixing Android deprecation errors,
- fixing minor warnings in Android-specific C++ code, and
- various minor fixes to the Android build scripts.

Closes #6321.

## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: scripts/android/files/build.gradle [hunk 1]
```diff
@@ -10,19 +10,25 @@ buildscript {
...
+	}
+}
+
 android {
-	compileSdkVersion 30
-	buildToolsVersion "30.0.3"
+	compileSdkVersion 34
 	ndkVersion "TW_NDK_VERSION"
 	defaultConfig {
 		applicationId "tw.DDNet"
```
Comment (heinrich5991) [inline scripts/android/files/build.gradle:28]: Should probably be `org.ddnet` nowadays.

File: src/engine/client/client.cpp [hunk 2]
```diff
@@ -4266,18 +4266,17 @@ int main(int argc, const char **argv)
...
+#else
 	bool Silent = false;
-
 	for(int i = 1; i < argc; i++)
 	{
 		if(str_comp("-s", argv[i]) == 0 || str_comp("--silent", argv[i]) == 0)
 		{
 			Silent = true;
 		}
 	}
```
Comment (heinrich5991) [inline src/engine/client/client.cpp:4279]: Could you move this `bool Silent` code to the usage of it instead?

**Review by heinrich5991 — COMMENTED** [review]
Have only skimmed it briefly so far.

File: scripts/android/cmake_android.sh [hunk 3]
```diff
@@ -1,17 +1,22 @@
 #!/bin/bash
 
-export ANDROID_HOME=~/Android/Sdk
-export MAKEFLAGS=-j32
+# $HOME must be used instead of ~ else cargo-ndk cannot find the folder
+export ANDROID_HOME=$HOME/Android/Sdk
+MAKEFLAGS=-j$(nproc)
```
Comment (ChillerDragon) [inline scripts/android/cmake_android.sh:5]: While `-j32` used to work on any mac (yes even on the ones with less than 32 threads) the `nproc` version only works for mac users that installed [coreutils](https://formulae.brew.sh/formula/coreutils#default) with brew. That should probably be mentioned as a dependency in the readme.

Comment (hrxi) [reply scripts/android/cmake_android.sh:5]: The script is likely only tested on Linux anyway.

Comment (Robyt3) [reply scripts/android/cmake_android.sh:5]: Is that not included in the macOS setup for ddnet overall? (which should be followed as the first step)

Our current README doesn't explain this either.

Comment (ChillerDragon) [reply scripts/android/cmake_android.sh:5]: Yea my comments on the pr that introduced nproc in the readme were ignored ._.

File: scripts/android/files/build.gradle (hunk 1 above)
Comment (Robyt3) [reply scripts/android/files/build.gradle]: Yeah, I changed the package name to `org.ddnet.client`.

File: src/engine/client/client.cpp (hunk 2 above)
Comment (Robyt3) [reply src/engine/client/client.cpp]: Done.

**Comment by Robyt3** [comment]
I removed 21a83c8421e270349b449ade1574f446aa6e748b as it's unnecessary, because this file doesn't actually exist in SDL2. It seems like it exists in SDL3 and the files got mixed up when I initially cloned and build with SDL3.

**Review by def- — APPROVED** [review]

**Comment by Robyt3** [comment]
Can we merge this so far or is there something immediate that should be addressed? I have another ~450 lines diff to clean up to build scripts further, do you prefer them in this PR or separately?

**Review by heinrich5991 — APPROVED** [review]
Thanks.

**Pull request merged**